### PR TITLE
WIP:  Source filtering and updated tests

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

File: tools/ztf_upload_avro.py [hunk 1]
```diff
@@ -660,6 +662,7 @@ def update_to(self, b=1, bsize=1, tsize=None):
     parser.add_argument('-c', '--clobber', action='store_true', default=False)
     parser.add_argument('--datadir', default='/tmp/ZTF')
     parser.add_argument('--workers', default=1)
+    parser.add_argument('--only_pure', action='store_true', default=False)
```
Comment (profjsb) [inline tools/ztf_upload_avro.py]: This should be True by default. We only want `pure` detections.

Comment (acrellin) [reply tools/ztf_upload_avro.py]: With `only_pure=True`, every single source gets skipped on my machine.

**Comment by dannygoldstein** [comment]
requesting this be merged

**Comment by stefanv** [comment]
Moved to #132

**Pull request closed without merging**